Create prototype for AST rewriter

ansley opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47216 Create prototype for AST rewriter**

Differential Revision: [D24687539](https://our.internmc.facebook.com/intern/diff/D24687539)

Commit:
Create prototype for AST rewriter

[ghstack-poisoned]

jamesr66a commented on torch/fx/symbolic_trace.py:
```diff
@@ -206,6 +206,10 @@ def forward(*args, **kwargs):
         return self.graph
 
+    def _proxy_placeholder(self, name: str, type_expr: Optional[Any] = None) -> Proxy:
+        return Proxy(self.create_node('placeholder', name, (), {}, type_expr=type_expr), self)
```
I don't think this is used anywhere. This and the import above should be deleted

jamesr66a commented on torch/fx/node.py:
```diff
@@ -20,6 +20,7 @@
     BaseArgumentTypes
 ]]
 
+
```
Please remove

jamesr66a reviewed (approved):
Please make some small changes before landing

Commit:
Update on "Create prototype for AST rewriter"




[ghstack-poisoned]

Commit:
Update on "Create prototype for AST rewriter"


Differential Revision: [D24687539](https://our.internmc.facebook.com/intern/diff/D24687539)

[ghstack-poisoned]

Commit:
Update on "Create prototype for AST rewriter"


Differential Revision: [D24687539](https://our.internmc.facebook.com/intern/diff/D24687539)

[ghstack-poisoned]

Commit:
Update on "Create prototype for AST rewriter"


Differential Revision: [D24687539](https://our.internmc.facebook.com/intern/diff/D24687539)

[ghstack-poisoned]
